crates/control/src/services/connectors.rs

```diff
@@ -1,17 +1,86 @@
 use std::path::Path;
 
+use anyhow::anyhow;
+use models::Object;
 use tokio::process::Command;
 
 use crate::config::settings;
+use crate::controllers::json_api::RawJson;
+use crate::models::connector_images::ConnectorImage;
+use crate::models::connectors::{Connector, ConnectorOperation};
+use crate::models::names::CatalogName;
+use crate::services::discovery::{DiscoverResponse, DiscoveredCatalog};
+use crate::services::subprocess::Subprocess;
 
-pub fn spec(image: &str) -> Command {
+#[derive(Debug, thiserror::Error)]
+pub enum ConnectorError {
+    #[error("could not execute the connector")]
+    ConnectorFailed(anyhow::Error),
+    #[error("failed to parse configuration")]
+    MalformedConfig(serde_json::Error),
+    #[error("could not parse the connector output")]
+    MalformedOutput(serde_json::Error),
+    #[error("connector does not support operation: `{0}`")]
+    UnsupportedOperation(ConnectorOperation),
+}
+
+pub async fn spec(image: &ConnectorImage) -> Result<RawJson, ConnectorError> {
+    // TODO: Swap `image.pinned_version()` out with `image.full_name()`?
+    let image_output = spec_cmd(&image.pinned_version())
+        .execute()
+        .await
+        .map_err(|e| ConnectorError::ConnectorFailed(anyhow!(e)))?;
+    serde_json::from_str(&image_output).map_err(ConnectorError::MalformedOutput)
+}
+
+fn spec_cmd(image: &str) -> Command {
     let mut cmd = Command::new("flowctl");
     cmd.arg("api").arg("spec").arg("--image").arg(image);
 
     cmd
 }
 
-pub fn discovery(image: &str, config_path: &Path) -> Command {
+/// Settings for controlling names generated as part of the discovered catalog.
+pub struct DiscoveryOptions {
+    /// The name given to the generated catalog itself. eg. `postgres` in `postgres.flow.json`
+    pub catalog_name: CatalogName,
```

Comment:
I'm stumbling over the naming here. Based on the comment, I think `resource_name` might be more appropriate. But I'm also not sure why it's a `CatalogName`. Is this also used for generating entity names?

Reply:
Yeah, I wasn't happy with the naming here either. I'd love to find a better term.

It's used in three places: 
1. generating the file name for `{name}.flow.json`
2. generating the file name for`{name}.config.json`
3. naming the capture itself, eg. `acmeCo/{name}`, which is naming a catalog entity. This one is why it's a `CatalogName`, though to be honest there may be valid `CatalogName`s which are not valid file names... :grimacing: 

Making all of these things all match each other is following the convention of `flowctl discover`'s output, but there may be better options? 

In any case, the best I could think of was that this is the name of the discovered catalog as a whole. That said, I'm not sure `resource_name` is quite it either though. We don't use the term `resource` for this concept either and I think it may raise just as many questions. Upon further reflection, it might be more accurately the `capture_name`? Any other ideas?

---

Slight aside from @travjenkins: by the time he calls this endpoint, he's already collected the `{name}` we're referring to here from the user. We could update the expectations of this endpoint to include the user-given name and use it here. I don't think it solves the question of what we call the variable here, but we wouldn't need to default it to the `connector.codename()`, we could use the value the user has already given us.

Reply:
The other thing that we discussed to today's UX/ControlPlane meeting was renaming `/discovery` to `/discovered_catalog`. We're all open to better naming on this front too, but it seemed more sensible to "create a discovered catalog from an image" than to "create a discovery from an image".